### DISCUSSION TITLE
When will Aggregators be implemented for the Post verb?

### DISCUSSION THREAD

**Discussion by Joseg-05** [ThreeMammals/Ocelot#1741] [Q&A] [answered]
Hello,

I really enjoy working with Ocelot, I was looking at Ocelot under the Request Aggregation section and on the very bottom of the doc is said that it is only implemented for the Get Verb, I was wondering when will they make for the other verbs which are Post, PUT, Delete?

Thank you

**Comment by raman-m** [comment, marked as answer]
Hi @Joseg-05 ! What's your name?
Thanks for your interest in Ocelot gateway!

From [Request Aggregation](../blob/develop/docs/features/requestaggregation.rst#gotchas--further-info) docs: 
> Aggregation only supports the GET HTTP verb.

OK, let's start debates...
The reason why the feature is implemented for GET verb is request body data block of HTTP protocol.
It is natural thing that aggregated upstream cannot have multiple request bodies because Ocelot request router will make N additional requests to aggregated downstreams also without a body.
In theory even GET request can have a body but by specification it is not recommended.
And if we will forward GET body of upstream then how to decide which exact downstream service should receive this body?
In theory, we can multiplex body to all aggregated downstreams, but do we have guarantee that all downstreams will respond with valid 2xx status? What to do in case of failure having invalid status? A lot of questions...

The same for POST request which has a body data block by specification. Which downstream do we have to forward this body? How to recognize this downstream? If forward this body to all downstreams, how to process responses, because most of the downstream will return bad status?
There are many pitfalls here, do you see?
This and another design problems were discussed by the authors of Request Aggregation feature in #79 
- #79 
Start reading from [this comment](../issues/79#issuecomment-290692800) to understand the request body problem.

Anyway, I would like to hear your design vision and probably to look at your draft solution. Please provide a feedback!
You are welcomed to create a draft PR to **develop** branch to see your designer's solution. Please fork Ocelot repo to your account and have some coding challenge to express your design vision.
But first, you need to review the current design of [Multiplexer](../tree/develop/src/Ocelot/Multiplexer) to understand the complexity of your proposal.

---
@RaynaldM @ggnaegi  Guys, what do you think? Please join the discussion if you have a time.

**Reply by RaynaldM** [reply]
Hi @Joseg-05, without revealing any secrets, do you have a use case? So that we can get a clearer idea

**Reply by ggnaegi** [reply]
@Joseg-05 What do you mean by Auth Server, identity provider, oidc?

**Comment by raman-m** [comment]
### Artifacts of discussion

**Source code**
- [Multiplexer](../tree/develop/src/Ocelot/Multiplexer)

**Docs**
- [Request Aggregation | Ocelot | Read the Docs](https://ocelot.readthedocs.io/en/latest/features/requestaggregation.html)
- [Request Aggregation | Ocelot RST source](../blob/develop/docs/features/requestaggregation.rst)

**Issues**
- #79 
- #298 

**PRs**
- #248 
- #310 
- #1330

**Reply by raman-m** [reply]
@Joseg-05, I've discovered an open PR #1389 that will introduce the POST verb. Would you be interested in prioritizing it?
